src/anomalib/data/image/mvtec_loco.py

```diff
@@ -0,0 +1,597 @@
+
+        The image is a torch tensor of shape (C, H, W) and the mask is a torch tensor of shape (H, W).
+
+        .. code-block:: python
+
+            print(dataset[0]["image"].shape, dataset[0]["mask"].shape)
+            # Output: (torch.Size([3, 256, 256]), torch.Size([256, 256]))
+    """
+
+    def __init__(
+        self,
+        task: TaskType,
+        transform: A.Compose,
+        root: Path | str = "./datasets/MVTec_LOCO",
+        category: str = "breakfast_box",
+        split: str | Split | None = None,
+    ) -> None:
+        super().__init__(task=task, transform=transform)
+
+        self.root_category = Path(root) / category
+        self.split = split
+
+    def _setup(self) -> None:
+        self.samples = make_mvtec_loco_dataset(
+            self.root_category,
+            split=self.split,
+            extensions=IMG_EXTENSIONS,
+            gt_merged_dir=GT_MERGED_DIR,
+        )
+
+    def __getitem__(self, index: int) -> dict[str, str | torch.Tensor]:
+        """Get dataset item for the index ``index``.
+
+        This method is mostly based on the super class implementation, with some different as follows:
+            - Using 'torch.where' to make sure the 'mask' in the return item is binarized
+            - An additional 'masks' is added, the non-binary masks with original size for the SPRO metric calculation
+        Args:
+            index (int): Index to get the item.
+
+        Returns:
+            dict[str, str | torch.Tensor]: Dict of image tensor during training. Otherwise, Dict containing image path,
+                target path, image tensor, label and transformed bounding box.
+        """
+        image_path = self._samples.iloc[index].image_path
+        mask_path = self._samples.iloc[index].mask_path
+        label_index = self._samples.iloc[index].label_index
+
+        image = read_image(image_path)
+        item = {"image_path": image_path, "label": label_index}
+
+        if self.task == TaskType.CLASSIFICATION:
+            transformed = self.transform(image=image)
+            item["image"] = transformed["image"]
+        elif self.task in (TaskType.DETECTION, TaskType.SEGMENTATION):
+            # Only Anomalous (1) images have masks in anomaly datasets
+            # Therefore, create empty mask for Normal (0) images.
+            mask = np.zeros(shape=image.shape[:2]) if label_index == 0 else cv2.imread(mask_path, flags=0)
+            mask = mask.astype(np.single)
+
+            transformed = self.transform(image=image, mask=mask)
+
+            item["image"] = transformed["image"]
+            item["mask_path"] = mask_path
+            # transform and binarize the mask
+            item["mask"] = torch.where(transformed["mask"] > 0, torch.tensor(1.0), torch.tensor(0.0))
+            # The non-binary masks with the original size for saturation based metrics calculation
+            item["masks"] = torch.tensor(mask)
```

Comment:
Just a quick question here. Did you verify that transforms don't by any chance change the values inside mask (like 244) as mask are usually all 255.

Reply:
At least for the current implementation, i have verified the transform did not change the value inside mask. However, it seems there is a plan to change the masking approach using tv_tensors `Mask` this PR https://github.com/openvinotoolkit/anomalib/pull/1706 . For that case, I haven't checked it.

Reply:
Okay, we'll need to keep this in mind for future, maybe we can cover this with tests?

Reply:
I added a unit test to check the output  the `mask` here:

https://github.com/openvinotoolkit/anomalib/blob/829289b80a1b34645d8f054965f8be4f88a71fc4/tests/unit/data/image/test_mvtec_loco.py#L34-L39